src/text/line.rs

```diff
@@ -585,36 +555,96 @@ impl Widget for Line<'_> {
 impl WidgetRef for Line<'_> {
     fn render_ref(&self, area: Rect, buf: &mut Buffer) {
         let area = area.intersection(buf.area);
+        let line_width = self.width();
+        if area.is_empty() || line_width == 0 {
+            return;
+        }
```

Comment:
`area.is_empty` is `O(1)` while `.width()` is `O(content.len())`. In other words: The written logic here still uses the expensive `line_width` even when the area is empty.
LLVM does not notice this, and I would prefer to explicitly write it like it was before.

Benchmark on Raspberry Pi 2:

```plaintext
line_render/Left/0      time:   [532.39 ns 532.45 ns 532.52 ns]
                        change: [+681.68% +683.74% +684.94%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 62 outliers among 1000 measurements (6.20%)
  30 (3.00%) high mild
  32 (3.20%) high severe
```

Reply:
Fixed in 10815513